### ISSUE TITLE
Unexpected display of metadata fields when creating dataset

### ISSUE THREAD

**Issue by tguilloteau** [IQSS/dataverse#8756]
**What steps does it take to reproduce the issue?**

* When does this issue occur?
In order to implement some modifications regarding the display of metadatablocks, such as 'Producer' and 'Project' when creating a dataset, we have made some modifications  on the citation.tsv file by assigning the value 'true' to metadata 'displayoncreate' for :
                          producer, producerAffiliation, producerName
                          project, projectAcronym, projectTitle

The purpose of this modification was to display these metadata fields from the creation of a new dataset.

* Which page(s) does it occurs on?
metadata display page when creating a new dataset

* What happens
The result, however, was somewhat different from expectations : Basically, the whole metadata blocks are displayed, even though some of  their fields have not been selected, meaning 'displayoncreate's value for these fields (producerLogoURL producerURL, producerAbbreviation and projectId, projectTask, projectURL, projectWorkPackage) is set to 'false'.

* To whom does it occur (all users, curators, superusers)?
All users

* What did you expect to happen?
We expected that the page would only display metadata fields whose 'displayoncreate' parameter is set to true.

**Which version of Dataverse are you using?**
5.3

**Any related open or closed issues to this bug report?**
None as we know so far
Some analyses have been conducted as far as Dataverse Java code is concerned. 
In all versions between 5.3 (our implemented version of Dataverse) and 5.10.1, it seems that the way metadata are displayed is set at metadatablock level, instead of field level.

Trying to properly choose which metadata field is to be displayed, is there any possibility or ways to discriminate these fields ?

**Screenshots:**

These folllowing screenshots show the actual display of producer and project metadatablock when we create a dataset.

* https://help.github.com/en/articles/file-attachments-on-issues-and-pull-requests
*
![project](https://user-images.githubusercontent.com/81553338/171339631-1f565e6c-3838-432c-a7ce-67b84feddb1d.JPG)
![producer](https://user-images.githubusercontent.com/81553338/171339652-c3bf3503-246c-4124-bcee-47486ef1b154.JPG)

**Comment by pdurbin** [comment]
@tguilloteau hi! I'm going to copy and paste what I just wrote at https://github.com/IQSS/dataverse/issues/8764#issuecomment-1145976407

First, I'd like to recommend AGAINST modifying citation.tsv. We may change it in the future. Here's the warning we put at https://guides.dataverse.org/en/5.10.1/admin/metadatacustomization.html

"Generally speaking it is safer to create your own custom metadata block rather than editing metadata blocks that ship with the Dataverse Software, because changes to these blocks may be made in future releases. If you’d like to make improvements to any of the metadata blocks shipped with the Dataverse Software, please open an issue at https://github.com/IQSS/dataverse/issues so it can be discussed before a pull request is made."

How do you feel about making a custom metadata block? Perhaps you could upload the .tsv to this issue and we could try it out if you continue to have trouble?

**Comment by pdurbin** [comment]
@tguilloteau thanks for you post at https://groups.google.com/g/dataverse-dev/c/x8sDRnnN6TQ/m/HRzkfgPQAAAJ (which I'm also pasting below).

Yes you have it right. A custom metadata block is the way to go, especially if the fields are specific to your institution. For Harvard Dataverse we have a number of custom metadata blocks for this or that department. Right now they are mostly at https://github.com/IQSS/dataverse/tree/v5.10.1/scripts/api/data/metadatablocks (the ones with "custom") but we're starting to move them to https://github.com/IQSS/dataverse.harvard.edu

I'll go ahead and close this issue. Please feel free to open a new issue if you're having problems.

Thanks.

--- 

 Hi Phil

Thanks so much for your reply.
I think I know have a better overview about what is going on as far as citation.tsv is concerned.
Basically, we understand that no metadatablock should be added through citation.tsv, we should work with a custom citation.tsv file as soon as we want to implement some changes (additionnal fields in our case)  with metadata.
We'll follow this pattern for further metadata modifications, and will test the solution for our 'project' metadatablocks as soon as possible, and let you know about results of our tests.

Thanks again for your time.

Best regards

Cordialement

Thierry  GUILLOTEAU
INRAE - Equipe Dataverse
